File: android/using-lazy-in-kotlin-to-bind-android-views/app/src/main/java/com/lordcodes/lazyviewbindingarticle/butterknife/PlanningActivity.java
package com.lordcodes.lazyviewbindingarticle.butterknife;

import android.os.Bundle;
import android.widget.ImageView;
import android.widget.TextView;

import com.lordcodes.lazyviewbindingarticle.R;

import androidx.appcompat.app.AppCompatActivity;
import butterknife.BindView;
import butterknife.ButterKnife;

public class PlanningActivity extends AppCompatActivity {
    @BindView(R.id.planning_text) TextView planningText;
    @BindView(R.id.app_icon) ImageView appIcon;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_planning);
        ButterKnife.bind(this);

        planningText.setText("Hello!");
    }
}

File: android/using-lazy-in-kotlin-to-bind-android-views/app/src/main/kotlin/com/lordcodes/lazyviewbindingarticle/lazy/PlanningActivity.kt
package com.lordcodes.lazyviewbindingarticle.lazy

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.lordcodes.lazyviewbindingarticle.R

class PlanningActivity : AppCompatActivity() {
//    private val planningText by lazy {
//        findViewById<TextView>(R.id.planning_text)
//    }
//    private val planningText by lazy(LazyThreadSafetyMode.NONE) {
//        findViewById<TextView>(R.id.planning_text)
//    }
//    private val planningText by bindView<TextView>(R.id.planning_text)
    private val planningText: TextView by bindView(R.id.planning_text)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_planning)

        planningText.text = "Hello!"
    }
}